setup-container.sh

```diff
@@ -1,176 +1,366 @@
+    log_info "read SSH public key: ${PUBKEY_FILE}"
+    SSH_PUBKEY="$(cat "${PUBKEY_FILE}")" || \
+    exit_failure "failed to read public SSH key: ${PUBKEY_FILE}"
 
-    if [[ "$HOST_USER_ID" != "$CONTAINER_USER_ID" ]]; then
-        echo "User ID mismatch between host and container"
-        exit 1
+    if [[ -f "${AUTHKEY_FILE}" ]]; then
+        grep -q "${SSH_PUBKEY}" "${AUTHKEY_FILE}" || echo "${SSH_PUBKEY}" >> "${AUTHKEY_FILE}"
+    else
+        cat "${PUBKEY_FILE}" > "${AUTHKEY_FILE}"
+        chmod 0644 "${AUTHKEY_FILE}"
     fi
 
-    if [[ "$HOST_GROUP_ID" != "$CONTAINER_GROUP_ID" ]]; then
-        echo "Group ID mismatch between host and container"
-        exit 1
+    TMP_DIR="$(mktemp -d)"
+    log_info "setup a temporary dir: ${TMP_DIR}"
+
+    log_info "prepare a Dockerfile template: ${TMP_DIR}/Dockerfile.j2"
+
+    cat <<'EOF' > "${TMP_DIR}/Dockerfile.j2"
+FROM {{ IMAGE_ID }}
+
+USER root
+
+# Group configuration
+RUN if getent group {{ GROUP_NAME }}; \
+then groupmod -o -g {{ GROUP_ID }} {{ GROUP_NAME }}; \
+else groupadd -o -g {{ GROUP_ID }} {{ GROUP_NAME }}; \
+fi
+
+# User configuration
+RUN if getent passwd {{ USER_NAME }}; \
+then usermod -o -g {{ GROUP_ID }} -u {{ USER_ID }} -m -d /home/{{ USER_NAME }} {{ USER_NAME }}; \
+else useradd -o -g {{ GROUP_ID }} -u {{ USER_ID }} -m -d /home/{{ USER_NAME }} -s /bin/bash {{ USER_NAME }}; \
+fi
+
+# Docker configuration
+RUN if getent group {{ DGROUP_NAME }}; \
+then groupmod -o -g {{ DGROUP_ID }} {{ DGROUP_NAME }}; \
+else groupadd -o -g {{ DGROUP_ID }} {{ DGROUP_NAME }}; \
+fi
+
+# Permissions configuration
+RUN usermod -a -G sudo {{ USER_NAME }}
+RUN usermod -aG {{ DGROUP_NAME }} {{ USER_NAME }}
+RUN echo '{{ USER_NAME }} ALL=(ALL) NOPASSWD:ALL' > /etc/sudoers.d/{{ USER_NAME }}
+RUN chmod 0440 /etc/sudoers.d/{{ USER_NAME }}
+
+# Environment configuration
+RUN chown -R '{{ USER_ID }}:{{ GROUP_ID }}' /var/AzDevOps
+RUN ln -s /var/AzDevOps/.local/ /root/.local
+RUN ln -s /var/AzDevOps/.local/ /home/{{ USER_NAME }}/.local
+
+# SSH/PASS configuration
+RUN sed -i -E 's/^#?PermitRootLogin.*$/PermitRootLogin yes/g' /etc/ssh/sshd_config
+RUN echo 'root:{{ ROOT_PASS }}' | chpasswd
+RUN echo '{{ USER_NAME }}:{{ USER_PASS }}' | chpasswd
+
+USER {{ USER_NAME }}
+
+ENV HOME=/home/{{ USER_NAME }}
+ENV USER={{ USER_NAME }}
+
+# Passwordless SSH access
+RUN mkdir -p ${HOME}/.ssh
+RUN chmod 0700 ${HOME}/.ssh
+RUN echo '{{ SSH_PUBKEY }}' >> ${HOME}/.ssh/authorized_keys
+RUN chmod 0600 ${HOME}/.ssh/authorized_keys
+
+WORKDIR ${HOME}
+
+EOF
+
+    cat <<EOF > "${TMP_DIR}/data.env"
+IMAGE_ID=${IMAGE_ID}
+DGROUP_NAME=${HOST_DGNAME}
+DGROUP_ID=${HOST_DGID}
+GROUP_ID=${HOST_GID}
+USER_ID=${HOST_UID}
+GROUP_NAME=${USER}
+USER_NAME=${USER}
+USER_PASS=${USER_PASS}
+ROOT_PASS=${ROOT_PASS}
+SSH_PUBKEY=${SSH_PUBKEY}
+EOF
+
+    log_info "generate a Dockerfile: ${TMP_DIR}/Dockerfile"
+    j2 -o "${TMP_DIR}/Dockerfile" "${TMP_DIR}/Dockerfile.j2" "${TMP_DIR}/data.env" || \
```

Comment:
"j2" may not be installed on host in case of configuring management docker from scratch.
It would be good to inform user how to fix it 
```
sudo apt install -y python3-pip
sudo pip3 install j2cli 
```

Reply:
done